typescript/expCalc.ts

```diff
@@ -0,0 +1,81 @@
+function Add(x,y)
+{
+	return x+y;
+}
+
+function Subtract(x,y)
+{
+	return x-y;
+}
+
+function Multiply(x,y)
+{
+	return x*y;
+}
+
+function TryDivide(x,y)
+{
+	if(x >= y)
+		return x/y;
+	else
+		return 0;
+}
+
+function getOperands(cleanInputString)
+{
+	let numArray: number[] = [];
+	for(let i in cleanInputString)
+	{
+		numArray[i] = parseFloat(cleanInputString[i]);
+	}
+	return numArray;
+}
+
+function calcValue (numArray, opArray)
+{
+	var total = numArray[0];
+	for(var i = 0 ; i < opArray.length; ++i)
+	{
+		switch (opArray[i])
+		{
+			case "+":
+				total = Add(total,numArray[i+1]);
+				break;
+			case "-":
+				total = Subtract(total,numArray[i+1]);
+				break;
+			case "*":
+				total = Multiply(total,numArray[i+1]);
+				break;								
+			case "/":
+				total = TryDivide(total,numArray[i+1]);
+				break;
+			case "":
+				break;								
+		}
+	}
+	return total;
+
+
+}
+
+function calcMain(stringExp)
+{
+	var operatorRegex = /[+-/*\/]/g;
+	var cleanRegex = /[^\+\-\*\/0-9]/g;
+	var modExp = stringExp.replace(cleanRegex,'');
```

Comment:
If I understand this statement invalid characters are removed from the user input, however, the user is unaware of the problem.